packages/core/src/messagingProtocols/mls/MLSService/MLSService.ts

```diff
@@ -134,6 +130,12 @@ export class MLSService extends TypedEventEmitter<Events> {
     }
   }
 
+  private async getCredentialType() {
+    return (await this.coreCryptoClient.e2eiIsEnabled(this.config.cipherSuite))
+      ? CredentialType.X509
+      : CredentialType.Basic;
+  }
```

Comment:
This fixes an issue where we need to dynamically change the creds type according to the e2ei enrollment status (this could not be static)